### PR TITLE
Add support for hdf5 1.12.1

### LINKED ISSUE #182: Add support for 1.12.1
Both the hdf5-sys's build.rs and new APIs if any.

### PR DESCRIPTION
**Pull request by mulimoen**
Fixes #182

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,7 @@
 
 - `Error` now implements `From<Infallible>`, which allows passing convertible
   extents (like tuples of integers) where `impl TryInto<Extents>` is required.
+- Support for `hdf5` version `1.12.1`.
```
Comment (aldanor) [inline CHANGELOG.md:9]:
```suggestion
- Support for HDF5 version 1.12.1.
```

(since we're mentioning upstream library version)

**Comment by aldanor** [comment]
This would have to be rebased on the docs branch, I think

**Comment by mulimoen** [comment]
`1.10.8` has only trivial changes and can be included too

**Comment by mulimoen** [comment]
I'll do a rebase after the `docs` PR is merged, should not be many (if any) changes required since this only affects `hdf5-sys`

**Comment by mulimoen** [comment]
This was rebased on the doc cfg branch

File: hdf5-sys/src/h5vl.rs [hunk 2]
```diff
@@ -22,3 +22,29 @@ extern "C" {
     ) -> hid_t;
     pub fn H5VLunregister_connector(vol_id: hid_t) -> herr_t;
 }
+
+#[cfg(hdf5_1_12_1)]
+#[repr(C)]
+#[derive(Copy, Clone, Debug, PartialEq)]
```
Comment (aldanor) [inline hdf5-sys/src/h5vl.rs:28]: Minor: `Eq` as well

**Comment by aldanor** [comment]
Minor: `offical` -> `official` in commit message

**Comment by aldanor** [comment]
Two minor nits aside, looks good!

**Review by aldanor — APPROVED** [review]

**Comment by mulimoen** [comment]
Thanks for the review @aldanor!

**Pull request merged**